### PR TITLE
Fix unused function parameters in the plugins functionality

### PR DESCRIPTION
**Pull request by freak82**
Use `/* name ATS_UNUSED */` for fully unused parameters.
Use `[[maybe_unused]]` attribute for conditionally unused parameters used in asserts.

This pull request is part of the effort for [removing](https://github.com/apache/trafficserver/issues/11377 ) the `-Wno-unused-parameter` warning suppression.

### CODE REVIEW

File: plugins/experimental/http_stats/http_stats.cc [hunk 1]
```diff
@@ -244,7 +244,7 @@ HeaderFieldStringSet(const HTTPStatsHttpHeader &http, const char *field_name, in
 }
 
 static TSReturnCode
-WriteResponseHeader(HTTPStatsRequest *trq, TSCont contp, TSHttpStatus status)
+WriteResponseHeader(HTTPStatsRequest *trq, TSCont /* contp ATS_UNUSED */, TSHttpStatus status)
```
Comment (JosiahWI) [inline plugins/experimental/http_stats/http_stats.cc:247]: Could we remove the `contp` parameter please?

File: plugins/experimental/icap/icap_plugin.cc [hunk 2]
```diff
@@ -256,7 +256,7 @@ handle_invalid_icap_behavior(TSCont contp, TransformData *data, const char *msg)
  *              icap eaders from response of icap server.
  */
 static int
-handle_icap_headers(TSCont contp, TransformData *data)
+handle_icap_headers(TSCont /* contp ATS_UNUSED */, TransformData *data)
```
Comment (JosiahWI) [inline plugins/experimental/icap/icap_plugin.cc:259]: Could we remove the `contp` parameter?

File: plugins/experimental/icap/icap_plugin.cc [hunk 3]
```diff
@@ -329,7 +329,7 @@ handle_icap_http_header(TransformData *data)
  *              downstream.
  */
 static int
-handle_read_http_body(TSCont contp, TransformData *data)
+handle_read_http_body(TSCont /* contp ATS_UNUSED */, TransformData *data)
```
Comment (JosiahWI) [inline plugins/experimental/icap/icap_plugin.cc:332]: Could we remove the `contp` parameter?

File: plugins/experimental/icap/icap_plugin.cc [hunk 4]
```diff
@@ -465,7 +465,7 @@ transform_connect(TSCont contp, TransformData *data)
 }
 
 static int
-transform_write_body(TSCont contp, TransformData *data)
+transform_write_body(TSCont /* contp ATS_UNUSED */, TransformData *data)
```
Comment (JosiahWI) [inline plugins/experimental/icap/icap_plugin.cc:468]: Could we remove the `contp` parameter?

File: plugins/experimental/icap/icap_plugin.cc [hunk 5]
```diff
@@ -707,7 +707,7 @@ transform_bypass(TSCont contp, TransformData *data)
  * Description: Buffer response body from origin server.
  */
 static int
-transform_buffer_os_resp(TSCont contp, TransformData *data)
+transform_buffer_os_resp(TSCont /* contp ATS_UNUSED */, TransformData *data)
```
Comment (JosiahWI) [inline plugins/experimental/icap/icap_plugin.cc:710]: Could we remove the `contp` parameter?

File: plugins/generator/generator.cc [hunk 6]
```diff
@@ -337,7 +337,7 @@ GeneratorWriteFailureResponse(GeneratorRequest *grq, TSHttpStatus status)
 }
 
 static TSReturnCode
-GeneratorWriteResponse(GeneratorRequest *grq, TSCont contp)
+GeneratorWriteResponse(GeneratorRequest *grq, TSCont /* contp ATS_UNUSED */)
```
Comment (JosiahWI) [inline plugins/generator/generator.cc:340]: Could we remove the `contp` parameter?

File: plugins/generator/generator.cc [hunk 7]
```diff
@@ -374,7 +374,7 @@ GeneratorWriteResponse(GeneratorRequest *grq, TSCont contp)
 }
 
 static TSReturnCode
-GeneratorPOSTResponse(GeneratorRequest *grq, TSCont contp)
+GeneratorPOSTResponse(GeneratorRequest *grq, TSCont /* contp ATS_UNUSED */)
```
Comment (JosiahWI) [inline plugins/generator/generator.cc:377]: Could we remove the `contp` parameter?

File: plugins/s3_auth/aws_auth_v4.cc [hunk 8]
```diff
@@ -612,8 +612,9 @@ getRegion(const StringMap &regionMap, const char *entryPoint, size_t entryPointL
  * @returns the string to sign
  */
 String
-getStringToSign(const char *entryPoint, size_t EntryPointLen, const char *dateTime, size_t dateTimeLen, const char *region,
-                size_t regionLen, const char *service, size_t serviceLen, const char *sha256Hash, size_t sha256HashLen)
+getStringToSign(const char * /* entryPoint ATS_UNUSED */, size_t /* EntryPointLen ATS_UNUSED */, const char *dateTime,
```
Comment (JosiahWI) [inline plugins/s3_auth/aws_auth_v4.cc:615]: Could we remove the `entryPoint` and `EntryPointLen` parameters? Also update the docstring.

File: plugins/slice/server.cc [hunk 9]
```diff
@@ -350,7 +350,7 @@ logSliceError(char const *const message, Data const *const data, HttpHeader cons
 }
 
 bool
-handleNextServerHeader(Data *const data, TSCont const contp)
+handleNextServerHeader(Data *const data, TSCont const /* contp ATS_UNUSED */)
```
Comment (JosiahWI) [inline plugins/slice/server.cc:353]: Could we remove the `contp` parameter?

File: plugins/statichit/statichit.cc [hunk 10]
```diff
@@ -296,7 +296,7 @@ HeaderFieldStringSet(const StaticHitHttpHeader &http, const char *field_name, in
 }
 
 static TSReturnCode
-WriteResponseHeader(StaticHitRequest *trq, TSCont contp, TSHttpStatus status)
+WriteResponseHeader(StaticHitRequest *trq, TSCont /* contp ATS_UNUSED */, TSHttpStatus status)
```
Comment (JosiahWI) [inline plugins/statichit/statichit.cc:299]: Could we remove the `contp` parameter?

File: plugins/xdebug/xdebug.cc [hunk 11]
```diff
@@ -447,7 +447,7 @@ InjectEffectiveURLHeader(TSHttpTxn txn, TSMBuffer buffer, TSMLoc hdr)
 }
 
 static void
-InjectOriginalContentTypeHeader(TSHttpTxn txn, TSMBuffer buffer, TSMLoc hdr)
+InjectOriginalContentTypeHeader(TSHttpTxn /* txn ATS_UNUSED */, TSMBuffer buffer, TSMLoc hdr)
```
Comment (JosiahWI) [inline plugins/xdebug/xdebug.cc:450]: Could we remove the `txn` parameter?

File: plugins/xdebug/xdebug_headers.cc [hunk 12]
```diff
@@ -119,7 +119,7 @@ class EscapeCharForJson
 ///////////////////////////////////////////////////////////////////////////
 // Dump a header on stderr, useful together with Dbg().
 void
-print_headers(TSHttpTxn txn, TSMBuffer bufp, TSMLoc hdr_loc, std::stringstream &ss)
+print_headers(TSHttpTxn /* txn ATS_UNUSED */, TSMBuffer bufp, TSMLoc hdr_loc, std::stringstream &ss)
```
Comment (JosiahWI) [inline plugins/xdebug/xdebug_headers.cc:122]: Could we remove the `txn` parameter?

**Review by JosiahWI — CHANGES_REQUESTED** [review]
Looks good. It'd be great if we could remove a few of these unused parameters as well.

**Comment by freak82** [comment]
@JosiahWI 
The pull request is updated with the requested changes.

**Review by JosiahWI — APPROVED** [review]
Thanks for the cleanup!

**Comment by JosiahWI** [comment]
[approve ci rocky]

**Review by cmcfarlen — APPROVED** [review]

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**